sdk/core/azure-core/CMakeLists.txt

```diff
@@ -23,13 +23,35 @@ find_package(Threads REQUIRED)
 
 # min version for `CURLSSLOPT_NO_REVOKE`
 # https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
-set(CURL_MIN_REQUIRED_VERSION 7.44)
+set(CURL_MIN_REQUIRED_VERSION 7.4)
 find_package(CURL ${CURL_MIN_REQUIRED_VERSION} CONFIG QUIET)
 if(NOT CURL_FOUND)
   find_package(CURL ${CURL_MIN_REQUIRED_VERSION} REQUIRED)
 endif()
 message("Libcurl version ${CURL_VERSION_STRING}")
 
+########################## Libcurl per-version found features. #################################
+#                                                                                              #
+#  Option `CURLSSLOPT_NO_REVOKE` will only be available for customers using version >= 7.44    #
+#                                                                                              #
+#  However, the Azure SDK for C++ can use an older version 7.4 witout supporting this option   #
+#                                                                                              #
+#    #     #              #            #           #          #            #                   #
+string(REPLACE "." ";" CURL_VERSION_LIST ${CURL_VERSION_STRING})
```

Comment:
What's the purpose of doing this?

Reply:
creates a list from the version string. 
CMake lists are semicolon-separated strings. This line creates a CMake list (think about it as an array) by splitting the original string on each dot.